src/libraries/System.Private.Xml/src/System/Xml/XmlConvert.cs

```diff
@@ -1203,47 +1278,41 @@ private static string[] AllDateTimeFormats
         {
             get
             {
-                if (s_allDateTimeFormats == null)
```

Comment:
this is potentially slower because CreateAllDateTimeFormats is not guaranteed to be inlined.